### PR TITLE
#272 Tests for news

### PR DESCRIPTION
**Pull request by MariaPadalka**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by DanyilTerentiev — CHANGES_REQUESTED** [review]
I would consider placing setups for mocks for example into private functions, which would reduce duplicated code

File: Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        //Arrange
+        var testNews = GetNews();
+
+
+        _mockRepository.Setup(x => x.NewsRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<DAL.Entities.News.News, bool>>>(), null))
+            .ReturnsAsync(testNews);
+        _mockRepository.Setup(x => x.SaveChangesAsync())
+            .ReturnsAsync(1);
+
+
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs:31]: spacing is odd.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    public async Task ShouldThrowExeption_IdNotExisting()
+    {
+        //Arrange
+        var testNews = GetNews();
+        var expectedError = $"No news found by entered Id - {testNews.Id}";
+
+        _mockRepository.Setup(x => x.NewsRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<DAL.Entities.News.News, bool>>>(), null))
+            .ReturnsAsync((DAL.Entities.News.News)null);
+
+
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs:57]: same here

File: Streetcode/Streetcode.XUnitTest/MediatRTests/News/UpdateNewsTest.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            // Assert
+            Assert.Multiple(() =>
+            {
+                Assert.True(result.IsSuccess);
+                Assert.IsType<NewsDTO>(result.Value);
+            });
+        }
+
+
+
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/News/UpdateNewsTest.cs:66]: same here

File: Streetcode/Streetcode.XUnitTest/MediatRTests/News/UpdateNewsTest.cs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                .Returns(testNews);
+            _mockMapper.Setup(x => x.Map<NewsDTO>(It.IsAny<DAL.Entities.News.News>()))
+                .Returns(testNewsDTO);
+            _mockRepository.Setup(x => x.ImageRepository.GetFirstOrDefaultAsync(
+                    It.IsAny<Expression<Func<Image, bool>>>(),
+                    It.IsAny<Func<IQueryable<Image>, IIncludableQueryable<Image, object>>>()
+                )).ReturnsAsync((Image)null);
+
+            _blobService.Setup(x => x.FindFileInStorageAsBase64(It.IsAny<string>()))
+                .Returns("base64Image");
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/News/UpdateNewsTest.cs:113]: also try to move setups to private methods, so you don't need to repeat your code.

**Review by Tysyatsky — COMMENTED** [review]
Check the code style and +1 to private methods.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+        [Fact]
+        public async Task ShouldThrowException_SaveChangesAsyncIsNotSuccessful()
+        {
+            // Arrange
+            var testNews = GetNews();
+            var expectedError = "Failed to delete news";
+            SetupMockRepositoryGetFirstOrDefault(testNews);
+            _mockRepository.Setup(x => x.SaveChanges()).Throws(new Exception(expectedError));
```
Comment (NadiaKishchuk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs:67]: you can set this value also in your mocking method passing the result of the method as a parameters

File: Streetcode/Streetcode.XUnitTest/MediatRTests/News/GetAllNewsTest.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        [Fact]
+        public async Task ShouldReturnSuccessfully_CorrectType()
+        {
+            // Arrange
+            SetupMockRepositoryGetAllAsync(GetNewsList());
+            _mockMapper.Setup(x => x.Map<IEnumerable<NewsDTO>>(It.IsAny<IEnumerable<DAL.Entities.News.News>>()))
+                .Returns(GetListNewsDTO());
+
```
Comment (NadiaKishchuk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/News/GetAllNewsTest.cs:33]: the same is here

**Review by KateYatsiuk — APPROVED** [review]

**Review by Dobriyr — APPROVED** [review]

**Review by Tysyatsky — APPROVED** [review]

**Pull request merged**